### DISCUSSION TITLE
SSH Session loggin option

### DISCUSSION THREAD

**Discussion by FishyZA** [1Remote/1Remote#785] [Ideas]
Would there be a option to save session data to a log file any time soon?
or is there one allready that i may have missed?

**Comment by VShawn** [comment]
Our ssh is achieved by kitty, and I don't know if kitty has this feature, you might want to check kitty's homepage?

**Comment by itagagaki** [comment]
Want to log session info locally, not server-side?
What type of info do you want?

I found config items something like that in the kitty GUI that 1Remote uses.

![image](https://github.com/user-attachments/assets/d8426176-997b-4de2-9200-c229597307b5)

But I have not actually tried it.

The function `logfopen` can also be found in the source code.
https://github.com/search?q=repo%3Acyd01/KiTTY%20logfopen&type=code

So it seems to have the feature, but I don't see how to make it work. `.ini` file? `.conf` file? Or environment variables? We'll be lucky if we can enable it with that. If it requires specifying command line options, that's a bit trickier.

**Comment by FishyZA** [comment]
i think just general info, like command input and outout, similar to what
putty does.
so that you can go back and look at specific date and see what or who broke
server

for some reason i think 1remote makes the ssh session looks nicer and much
smoother.
will have to play around with that, but at this point it seems like a
little too much effort :)

however, you screenshot is exactly what i was looking for, could / can not find it in 1remote directly though.

**Comment by itagagaki** [comment]
Okay, I figured out how to do it.
KiTTY's log output feature can be used without modifying 1Remote.

For example, create a session file like the one below.

```
LogType\3\
LogFileClash\-1\
LogFlush\1\
LogFileName\test.log\
```

Specify this file in the "KiTTY Session" in the 1Remote host settings.
Then, in this example, logs are written to the test.log file.

`LogType` is likely to be the following.
\# | what
--- | ---
0 | None
1 | Printable output
2 | All session output
3 | SSH packets
4 | SSH packets and raw data

`LogFileClash` is likely to be the following.
\# | What to do if the log file already exists
--- | ---
-1 | Ask the user
0 | Overwrite
1 | Append

`LogFlush` is likely to be the following.
\# | Flush log file frequently
--- | ---
0 | No
1 | Yes

**Reply by VShawn** [reply]
so you need inject external parameters by using a Kitty session file?

![image](https://github.com/user-attachments/assets/dbbc26f9-fccb-43f6-b0e3-ef649b418263)

**Reply by itagagaki** [reply]
> so you need inject external parameters by using a Kitty session file?

Exactly. That will work fine :)

**Reply by VShawn** [reply]
I always thinking this session file is too hardcore, maybe examples should be given on our official website.

**Reply by itagagaki** [reply]
It would be good to have a GUI to generate KiTTY session files. Preferably one that covers all KiTTY configuration items. It would be very nice if it could be built into 1Remote. But there is no need for that, a standalone GUI application would suffice. Maybe something like that already exists. I'll have to look for it.

**Reply by VShawn** [reply]
> Maybe something like that already exists. I'll have to look for it.

I dont think so, I once sought for it, failed.

**Reply by itagagaki** [reply]
I did a bit of searching, but as you say, I couldn't find any off-the-shelf tools.
Nor could I find any accurate documentation on the KiTTY session configuration file.
So it looks like I'll have to analyse the source code (as I did yesterday) and write documentation.
It will take some time, but I'll do it bit by bit.
Like this.
[kitty_settings.zip](https://github.com/user-attachments/files/18196096/kitty_settings.zip)

**Reply by itagagaki** [reply]
I made a documentation about KiTTY session files. Feel free to copy and use it.
Please change `kitty_portable.exe` to `kitty_portable_1Remote.exe` and supplement where it is located.
Also, add a description of the items that 1Remote overrides.

[How to prepare a KiTTY configuration file](https://itagaki.eek.jp/docs/KiTTY/kitty_settings.html)
